docs/contributor/03-11-hap-rules.md

```diff
@@ -0,0 +1,293 @@
+
+## Rule Evaluation
+
+HAP evaluates a set of rules to determine what labels to use when querying secret bindings.
+Rule is evaluated during cluster provisioning.
+Rule entries are analyzed one by one.
+Physical attributes from every entry are compared with values from input, if they are the same the rule entry is considered triggered.
+If there is more than one triggered rule then only one is selected in accordance to priority rules described in the [Priority](#priority) section.
+If there are no rule entries trigered then the error is returned. No fallback behaviour is defined in such case.
+
+## Uniqnuess & Priority
+
+Only one rule can be triggered.
+
+If more than one rule entry match the request than only one is filterd out and applied. 
+
+The process of selecting the best matching rule is based on a rule uniqnues and priority.
+
+Rule entry uniqnuess is determined by its plan and input parameters with its values (identification attributes) not including input parameters with `*`. 
+
+Output parameters are not taken into account when establishing rule entry uniqnuess. For example the following rule will fail on startup because every skr can be matched by both rules:
+
+```
+hap:
+  rule: 
+    - gcp 
+    - gcp -> S # invalid entry, output attributes does not take part into uniqnuess check
+    - gcp(PR=*) # invalid entry, both can be applied to the same skr
+```
+
+The similary behaviour occurs when the rule defined the same attributes values together with `*` value.
+
+```
+hap:
+  rule: 
+    - gcp(HR=europe-west3) 
+    - gcp(PR=*, HR=europe-west3) # invalid rules, both can be applied to the same skr
+```
+
+Rule configuration must contain only unique entries in the scope of that list.
+If rule configuration contains duplicated rule entries an error during KEB startup is returned.
+
+Rule entry priority is selected by sorting all the rules entries that apply to the request by the number of indentification attributes they contain. 
+For example, a rule with no attributes and only plan has lower priority then a rule with the same plan but a platform region attribute (`gcp` < `gcp(PR=cf-sa30)`).
+
+After sorting the entry that specifies most attributes (hence is more specific) is selected. 
+Input attributes with value `*` is not taken into account when calculating priority.
+
+The following example shows the priority of the rules from the general rule to more specific ones:
+
+```
+aws -> S # search labels: hyperscalerType: aws, shared: true
+aws(PR=cf-eu11) -> EU # search labels: hyperscalerType: aws_cf-eu11, euAccess: true
+aws(PR=cf-eu11, HR=westeu) -> EU, S # search labels: hyperscalerType: aws_cf-eu11_westeu, shared: true, euAccess: true
+```
+
+## Search Labels
+
+The HAP stores credentials for the hyperscaler accounts that have been set up in advance in Kubernetes Secrets that the Secret Bindings, that KEB searches for, point to. The SecretBindings are labeld with **hyperscalerType**, **shared** and **euAccess** labels. 
```

Comment:
I would consider rephrasing since this sentence is quite complicated and not easy to comprehend.

Reply:
Corrected.